flink-runtime/src/main/java/org/apache/flink/runtime/operators/Driver.java

```diff
@@ -87,4 +87,6 @@
 	 * @throws Exception Exceptions may be forwarded.
 	 */
 	void cancel() throws Exception;
+
+	void resetForIterativeTasks() throws Exception;
```

Comment:
Ok. Let me update the code accordingly. Of course it is simple then. My only concern was I was not sure whether every driver can be made of type ResettableDriver when it was a special case for certain drivers.